plugins/woocommerce/src/Caches/OrderCountCacheService.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Caches;
+
+use WC_Order;
+use Automattic\WooCommerce\Utilities\OrderUtil;
+
+/**
+ * A service class to help with updates to the aggregate orders cache.
+ */
+class OrderCountCacheService {
+
+	/**
+	 * OrderCountCache instance.
+	 *
+	 * @var OrderCountCache
+	 */
+	private $order_count_cache;
+
+	/**
+	 * Class initialization, invoked by the DI container.
+	 *
+	 * @internal
+	 */
+	final public function init() {
+		$this->order_count_cache = new OrderCountCache();
+		add_action( 'woocommerce_new_order', array( $this, 'update_on_new_order' ), 10, 2 );
+		add_action( 'woocommerce_before_delete_order', array( $this, 'update_on_delete_order' ), 10, 2 );
+		add_action( 'woocommerce_order_status_changed', array( $this, 'update_on_order_status_changed' ), 10, 4 );
+	}
+
+	/**
+	 * Update the cache when a new order is made.
+	 *
+	 * @param int      $order_id Order id.
+	 * @param WC_Order $order The order.
+	 */
+	public function update_on_new_order( $order_id, $order ) {
+		if ( ! $this->order_count_cache->is_cached( $order->get_type() ) ) {
```

Comment:
I removed hydration of statuses in these callbacks since it's possible these are indirectly called via the frontend (e.g., checkout), which would cause long delays in processing orders for customers.